home-manager/home-manager

```diff
@@ -87,14 +86,16 @@ function doBuild() {
 }
 
 function doSwitch() {
-    local wrkdir
-    wrkdir="$(mktemp -d)"
+    local wrkdir="$(mktemp -d)"
```

Comment:
This use of `local` on a separate line is actually intentional due to some odd behavior in the shell:

```
$ cat <<EOF > shellWeirdness 
set -e
function f() {
  local x=\$(false)
  echo err \$?
}            
f
EOF
$ cat <<EOF > shellWeirdness2
set -e
function f() {
  local x
  x=\$(false)
  echo err \$?
}
f
EOF
$ sh shellWeirdness
err 0
$ sh shellWeirdness2
$ echo $?
1
$ 
```
I got this from [shellsheck](https://github.com/koalaman/shellcheck/wiki/SC2155) and basically I want the script to error out if, e.g., `mktemp` fails. This is the reason for